src/pages/signin/SAMLSignInPage/index.tsx

```diff
@@ -1,21 +1,28 @@
 import React, {useEffect} from 'react';
-import {withOnyx} from 'react-native-onyx';
+import {useOnyx} from 'react-native-onyx';
 import SAMLLoadingIndicator from '@components/SAMLLoadingIndicator';
+import {fetchSAMLUrl} from '@libs/LoginUtils';
 import CONFIG from '@src/CONFIG';
 import ONYXKEYS from '@src/ONYXKEYS';
-import type {SAMLSignInPageOnyxProps, SAMLSignInPageProps} from './types';
 
-function SAMLSignInPage({credentials}: SAMLSignInPageProps) {
+function SAMLSignInPage() {
+    const [credentials] = useOnyx(ONYXKEYS.CREDENTIALS);
     useEffect(() => {
-        window.location.replace(`${CONFIG.EXPENSIFY.SAML_URL}?email=${credentials?.login}&referer=${CONFIG.EXPENSIFY.EXPENSIFY_CASH_REFERER}`);
+        const body = new FormData();
+        body.append('email', credentials?.login ?? '');
+        body.append('referer', CONFIG.EXPENSIFY.EXPENSIFY_CASH_REFERER);
+
+        fetchSAMLUrl(body).then((response) => {
```

Comment:
If this fails, what will the user see? 👀

Reply:
Ooh although actually... it think that might not be handled - let me add a catch that just drops them back on the main sign in page